README.md

```diff
@@ -1,115 +1,108 @@
-        3. Reduction: echelon, canonical, triangular decomposition
-
-        4. Row and column Transformation operations
-        5. Decomposition: Triangular, Symmetric, hermitian decomposition
-        6. Matrix Geometry: eigenvalues, modal, spectral, nullspace
-                           algebraic multiplicity, geometric multiplicity
-                           of a Matrix
-        7. Differentiation
-        8. Solving and comparing Matrices
-
-        The matrix module or Matrix Class/datatype comes in two implementations.
-        as Matrix or Matrix_
-        Both can be import from engpy
-        >>> from engpy import Matrix
-        or
-        >>> from engpy import Matrix_
-        
-        See the Matrix_doc file or engpy Arrays documentation to learn its usage
-        
-
-    3. Vector
-        This data type holds the keys to vector analysis.
-        1. Simple Vector Algebra: Addition, Subtraction, Substitution, Modulus
-                                  of Vectors
-        2. Angles between Vectors
-        3. Multiplication of Vector: Dot, and scalar product
-        4. Vector Calculus: Differentiation and Integration
-        5. Vector Operations: Tangents, normals, grad, directional derivatives,
-                              div, curl
-        6. Validating properties: solenoidal, irrotational, coplanar, orthogonality
-        7. Scalar, Vector Triple product
-
-        Vector Class can also be imported from engpy
-        >>> from engpy import Vector
-
-
-    Note that All these three datatypes works with python operators, +, -, /, *, ~
-    e.g MatObj1 + MatObj2
-'''        
-        
+# EngPy
+
+EngPy, Python for Engineers; Python for Engineering, a free and open-source Python library for Engineering computations.
+
+EngPy is targeted at handling most engineering problems like calculus, transforms, graphs, complex algebraic expressions, Matrices manipulation, vector analysis, analyzing signals.
+
+The library consists of three major datatypes:
+
+## 1. Expr
+This is the core datatype of EngPy. Most other parts of the library are based on this class.
+It is responsible for any algebraic manipulations.
+
+1.  Simple Algebra: Addition, Subtraction, Multiplication, Division Substitution of Expressions; change of subject of formula, clear brackets, and fractions.
+2.  Calculus: linear and partial differentiation, integration, gradient
+3.  Trigonometry
+4.  logarithmic Expressions
+5.  Transforms: Laplace, Z-Transforms
+6.  Visualization: Graphs
+7.  Table of Values
+8.  Complex Number Manipulation
+9.  Solving Expressions
+10. Support the engpy AI Implementation for manipulating expressions
+
+`Expr` can be imported from `engpy`:
+
+```python
+>>> from engpy import Expr
+```
+
+See the doc file or `Expr`'s documentation.
+
+To interact with `Expr` instances as discrete objects, use the `engpy.interface` module.
+The `interface` module bridges between the `Expr` class and `Expr` datatypes.
```

Comment:
@miracle5284 Couldn't figure out what this sentence should mean...

Reply:
Just to inform the users of the options of interacting with either Expr directly bypassing strings or  the interface module by import functions

Reply:
I guess you meant to say
> Just to inform the users of the options of either interacting with `Expr` directly by passing strings or by using functions provided by the `interface` module.

If that's it, then the sentence on line 33 should definitely be changed... that was what i was trying to point out.
"`Expr` class" and "`Expr` datatype" are _technically_ the same thing.

I suggest it be changed to:
> The `interface` module provides utilities for interacting with `Expr` instances as discrete objects.

Of course, it's been merged now, but the change can be made along with some other later.
Next time, I'll make such PRs to be drafts.